src/main/resources/public/jsx/appmanagement/service-settings.jsx.js

```diff
@@ -86,12 +86,7 @@ var Service = React.createClass({
             links.push(
                 <span key={this.props.service.service.id + '-settings'} className="pull-right action-icon" onClick={this.settings}>
                     <i className="fa fa-cog fa-lg"></i>
-                </span>
-            );
-        }
-
-        if (!this.state.saved_service.service.visible && this.props.status !== 'STOPPED') {
-            links.push(
+                </span>,
```

Comment:
Yes it is necessary.
But there is a refactoring code and the 2 "<span>" are in the same "link.push(" with the same condition and only separate by a comma.